addons/account/models/account_move.py

```diff
@@ -3878,13 +3882,17 @@ def _valid_field_parameter(self, field, name):
         # I can't even
         return name == 'tracking' or super()._valid_field_parameter(field, name)
 
-    def unlink(self):
-        moves = self.mapped('move_id')
 
+    @api.ondelete(at_uninstall=False)
+    def _moves_posted(self):
+        moves = self.mapped('move_id')
         # Prevent deleting lines on posted entries
         if not self.env.context.get('force_delete', False) and any(m.state == 'posted' for m in moves):
             raise UserError(_('You cannot delete an item linked to a posted entry.'))
```

Comment:
Could access move_id lazily in this case, that way if `force_delete` is set we don't even have to read the moves.